### PR TITLE
Normalize and scale load profile, refactor (mostly heating) load code

### PR DESCRIPTION
**Pull request by Bill-Becker**
- Add new input for each load called `normalize_and_scale_load_profile_input` which takes the input custom load profile (e.g. `ElectricLoad.loads_kw`), normalizes it to timestep_energy/yearly_energy for each timestep, and then applies and scales that normalized profile to the input annual monthly energy values.
  - Before, we could only scale annual or monthly energy inputs to the CRB profiles. 
  - This leverages the CRB calculations for applying and scaling annual and monthly energy to the load profile.
  - This will be especially helpful to use along with the upcoming Load Profile Builder feature in the web tool
- Put all of the default `annual_[energy]` data for CRBs by city/zone into .json files in the `data/load_profiles` folder, and then load that data within the load profile constructor functions (`BuiltIn[Load]`) where they lived before (hard to navigate those load file with all that data).
- Consolidate all of the duplicate code for processing the inputs and building structs for the three different types of heating loads into a single `HeatingLoad` and `BuiltInHeatingLoad` functions (3x2=6 functions down to 2) with a new argument of `(heating_)load_type`.
- Update the `simulated_load` function for improved validation flow and adding the `normalize_and_scale_load_profile_input` option

### CODE REVIEW

**Comment by Bill-Becker** [comment]
@zolanaj This is needed for end-of-December milestone in the web tool, if you're able to do at least a fatal flaw review soon. Thanks!

File: test/runtests.jl
Comment (zolanaj) [inline test/runtests.jl:3009]: > these test monthly builds but not annual, I'll add a set.

see ddcbec3393ffe3a8cc3a4fc0da114843b44f1c57

File: src/core/heating_cooling_loads.jl
Comment (zolanaj) [inline src/core/heating_cooling_loads.jl:68]: with this naming convention, should we rename the `dhw_load` object in the scenario object to `domestic_hot_water_load`?  If you agree, I may add that to the other const refactoring in a separate PR so it doesn't block progress on this.  That way, load types would stay consistent

File: test/test_with_xpress.jl
Comment (zolanaj) [inline test/test_with_xpress.jl:1554]: It's good that we're keeping up with this, but is it worth a conversation revisiting whether we keep these?  I think that with battery degaradation updated, all indicator constraints have been refactored for HiGHS compatibility.

**Review by zolanaj — APPROVED** [review]
Thanks @Bill-Becker!  This looks great, and I tested a fair amount and didn't see any concerns.  I have a few suggestions, some of which are in the form of additional commits that you can revert if you disagree. The rest of the comments are more for discussion and potential further refactoring.  This is good to merge either way (assuming the merged to update with develop passes).

File: docs/src/reopt/inputs.md
Comment (Bill-Becker) [reply docs/src/reopt/inputs.md]: Yeah, that makes sense.

File: src/core/simulated_load.jl
Comment (Bill-Becker) [reply src/core/simulated_load.jl]: Updated to uppercase names for these consts in 79078bcb2b2effefc7345e6f21d208247d6bc455.

File: test/runtests.jl
Comment (Bill-Becker) [reply test/runtests.jl:3009]: Thanks!

File: src/core/heating_cooling_loads.jl
Comment (Bill-Becker) [reply src/core/heating_cooling_loads.jl:68]: I agree with that change. But yeah I just looked at the "search and replace" and it's a lot especially if we want to include updated the names of the results from e.g. "thermal_to_dhw_load..." to "thermal_to_domestic_hot_water_load...". The results would require some renames on the API side as well. For now, let's punt and put on the TODO list.

File: test/test_with_xpress.jl
Comment (Bill-Becker) [reply test/test_with_xpress.jl:1554]: Right, I agree we could remove "test_with_xpress.jl". Let's punt on that for now though.

**Comment by Bill-Becker** [comment]
> Thanks @Bill-Becker! This looks great, and I tested a fair amount and didn't see any concerns. I have a few suggestions, some of which are in the form of additional commits that you can revert if you disagree. The rest of the comments are more for discussion and potential further refactoring. This is good to merge either way (assuming the merged to update with develop passes).

Thanks for the review @zolanaj! I agreed with all of your commits and made a couple of updates from your comments. Waiting on tests to pass after those updates, then I will merge.

**Pull request merged**
